build.psm1

```diff
@@ -723,7 +723,23 @@ function New-PSOptions {
         $Top = [IO.Path]::Combine($PSScriptRoot, "src", "System.Management.Automation")
     }
 
-    return @{ Top = $Top;
+    $RootInfo = @{RepoPath = $PSScriptRoot}
+
+    # the valid root is the root of the filesystem and the folder PowerShell
+    $RootInfo += @{ValidPath = Join-Path -Path ([system.io.path]::GetPathRoot($RootInfo.RepoPath)) -ChildPath 'PowerShell' }
```

Comment:
A minor comment about the way you use Hashtable: in this scenario it's better to write script like 
```
$RootInfo['ValidPath'] = Join-Path -Path ([system.io.path]::GetPathRoot($RootInfo.RepoPath)) -ChildPath 'PowerShell'
$RootInfo['IsValid'] = $true
$RootInfo['Warning'] = "Please ensure you repo is at the root of the file system and named 'PowerShell' (example: '$($RootInfo.ValidPath)'), when building and packaging for release!"
```
It's much less expensive than adding two hashtables.